libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/utils/DirectionsRouteMissingConditionsCheckTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.mapbox.navigation.base.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import junit.framework.Assert.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.net.URL
+
+class DirectionsRouteMissingConditionsCheckTest {
+
+    private val exclusiveQueries = mapOf(
+        "engine" to "electric",
+        "key1" to "value1",
+    )
+
+    @Before
+    fun setup() {
+        mockkObject(QueriesProvider)
+        every { QueriesProvider.provideQueries() } returns exclusiveQueries
+    }
+
+    @After
+    fun cleanUp() {
+        unmockkObject(QueriesProvider)
+    }
+
+    @Test
+    fun cases() {
+        listOf(
+            CasesWrapper(
+                "Not restricted key-value pair",
+                "https://test.com/?key0=value0",
+                CasesWrapper.Result.Success
+            ),
+            CasesWrapper(
+                "Restricted key-value pair",
+                "https://test.com/?key1=value1",
+                CasesWrapper.Result.Error("[(key1=value1)]")
+            ),
+            CasesWrapper(
+                "Restricted key-value pair with other valid queries: engine=electric",
+                "https://test.com/?key0=value0&engine=electric&key2=value2",
+                CasesWrapper.Result.Error("[(engine=electric)]")
+            ),
+            CasesWrapper(
+                "Restricted key and valid value: engine=nonelectric",
+                "https://test.com/?engine=nonelectric",
+                CasesWrapper.Result.Success
+            ),
+            CasesWrapper(
+                "Valid key and restricted valid value: machine=electric",
+                "https://test.com/?key0=value0&machine=electric",
+                CasesWrapper.Result.Success
+            ),
+            CasesWrapper(
+                "A few restricted key and valid values: engine=electric " +
+                    "and key1=value1",
+                "https://test.com/?key0=value0&engine=electric&key1=value1",
+                CasesWrapper.Result.Error("[(engine=electric);(key1=value1)]")
+            ),
+        ).forEach { (description, url, result) ->
+            val directionRoute = mockk<DirectionsRoute> {
+                every { routeOptions()?.toUrl(any()) } returns URL(url)
+            }
+
+            try {
+                DirectionsRouteMissingConditionsCheck.checkDirectionsRoute(directionRoute)
+            } catch (e: IllegalStateException) {
```

Comment:
Don't you have `assertThrows` or sth?
It's quite complex now. For example, if `checkDirectionsRoute` does not throw but we expect error, will the test fail? I think not.

Reply:
> It's quite complex now. For example, if checkDirectionsRoute does not throw but we expect error, will the test fail?

it throws actually additional exception, see
```
                    CasesWrapper.Result.Success -> {
                        throw IllegalStateException("Illegal state")
                    }
```

Changed to `assertThrows `,  we don't check the message, but believe we should be good, all data is there.